### DISCUSSION TITLE
Chaos Mode

### DISCUSSION THREAD

**Discussion by philsturgeon** [stoplightio/prism#471] [Ideas]
**User story.**
As a client developer, I can enable chaos mode, which will make the mock act like an unstable server. Latency will vary, sometimes 50X errors will come back, and things will generally be a mess.

**Is your feature request related to a problem?**
Production APIs [trip, stumble, or fall over entirely](https://apisyouwonthate.com/blog/taking-a-timeout-from-poor-performance), and we need to help clients get ready for that. 

Clients can set up toxiproxy for their mocks to enable this functionality, and maybe we can work with that, but maybe we can implement something a bit more simplistic that does not require running a toxiproxy server.

**Describe the solution you'd like**

TBD

**Additional context**

Similar implementations...

**Comment by gareth-a-jones** [comment]
Not just unstable, but also hyper-diverse withing the spectrum of possible replies 

Firstly that the specification defines,
```
empty arrays, 
single item arrays, 
huge volume arrays, 
empty properties, 
long properties
```

Secondly that the unspecified implicit contract defines (maybe using custom extensions?)
```
paged data, one and many pages,

random DNS addresses for hypermedia unrelated to base address of service, 

random auth token length (if prism mocks oauth too, or maybe require certain lookahead on expiry time if not?) 

random authZ roles required,

random refusing of query param combos or query param values.
```

Not sure 'random' really means 'random' in those phrases above.

**Comment by philsturgeon** [comment]
Mockit has a boolean for Monkey Mode https://mockit.netlify.com/docs/features/chaos-monkey

**Comment by XVincentX** [comment]
Yeah, this is really a cool feature we could develop and bring a lot of value. 

We probably just need to define what "chaos" mean; returning errors (which is ok) but which errors? 404? 500? According to the ones defined in the document? Or even return 200 with a different payload? Missing fields?

**Comment by philsturgeon** [comment]
I was sharing mockit as an example to help define that :)

**Comment by lornajane** [comment]
We're considering recommending something like this to our users for testing. +1 for chaos mode!
